src/tools/rust-analyzer/crates/proc-macro-srv/src/dylib/version.rs

```diff
@@ -123,9 +123,8 @@ pub fn read_version(dylib_path: &AbsPath) -> io::Result<String> {
     let version = u32::from_be_bytes([dot_rustc[4], dot_rustc[5], dot_rustc[6], dot_rustc[7]]);
     // Last supported version is:
     // https://github.com/rust-lang/rust/commit/b94cfefc860715fb2adf72a6955423d384c69318
-    let (snappy_portion, bytes_before_version) = match version {
-        5 | 6 => (&dot_rustc[8..], 13),
-        7 | 8 => {
+    let (mut metadata_portion, bytes_before_version) = match version {
+        8 => {
```

Comment:
When dylibs switched from compressed to uncompressed metadata the metadata version was set to 8, so anything older than version 8 can't be read anyway without support for compressed metadata.